include/wx/charts/wxdoughnutchart.h

```diff
@@ -27,19 +27,18 @@
 
 #include "wxdoughnutandpiechartbase.h"
 #include "wxdoughnutchartoptions.h"
+#include "wxchartobservers.h"
 
 /// Data for the wxDoughnutChartCtrl control.
-class wxDoughnutChartData
+class wxDoughnutChartData : public wxChartObservableValue<wxVector<wxChartSliceData>>
 {
 public:
     wxDoughnutChartData();
 
-    void AppendSlice(const wxChartSliceData &slice);
-
     const wxVector<wxChartSliceData>& GetSlices() const;
-
-private:
-    wxVector<wxChartSliceData> m_slices;
+    void AppendSlice(const wxChartSliceData &slice);
+    void UpdateData(const wxVector<wxChartSliceData> &data);
```

Comment:
Last comment and I'll merge the review. Can we rename these UpdateSlice and AddSlices, just to keep the naming consistent.